pkg/apps/packagekit.js

```diff
@@ -73,7 +73,7 @@ export function install(name, progress_cb) {
                 progress.base = 1;
                 progress.range = 99;
 
-                return PK.cancellableTransaction("InstallPackages", [0, [pkgid]], progress.progress_reporter)
+                return PK.install_missing_packages({ missing_ids: [pkgid] }, progress.progress_reporter)
```

Comment:
Hmm, I'd rather there be a new `PK.install_packages` method. `install_missing_packages` is only meant to  be used together with `check_missing_packages`, and using it with explicitly constructed objects instead of the return value of check_missing_packages is too hackish, imo.

The new install_packages can of course call install_missing_packages in its implementation, if that makes sense. But it looks like install_missing_packages should actually call the new install_packages.